src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -21,180 +21,180 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
                 case ProgressStatus.Stopped:
-                    StartedOrStopped(started: false);
+                    Stopped();
                     break;
                 case ProgressStatus.Evaluating:
                 case ProgressStatus.Paused:
-                    _lastProgressStatus = progressData.Status;
-                    ProgressUpdated();
+                    ChangeProgress(GetMessage(progressData, _lastProgressCount));
                     break;
                 default:
                     throw ExceptionUtilities.UnexpectedValue(progressData.Status);
             }
         }
 
-        private void ProgressUpdated()
+        private static string GetMessage(ProgressData progressData, int pendingItemCount)
         {
-            // we prefer showing evaluating if progress is flipping between evaluate and pause
-            // in short period of time.
-            var forceUpdate = _lastShownProgressStatus == ProgressStatus.Paused &&
-                              _lastProgressStatus == ProgressStatus.Evaluating;
-
-            var current = DateTimeOffset.UtcNow;
-            if (!forceUpdate && current - _lastTimeReported < s_minimumInterval)
-            {
-                // make sure we are not flooding UI. 
-                // this is just presentation, fine to not updating UI right away especially since
-                // at the end, this notification will go away automatically
-                // but to make UI to be updated to right status eventually if task takes long time to finish
-                // we enqueue refresh task.
-                EnqueueRefresh();
-                return;
-            }
-
-            _lastShownProgressStatus = _lastProgressStatus;
-            _lastTimeReported = current;
-
-            ChangeProgress(_taskHandler, GetMessage());
-
-            string GetMessage()
-            {
-                var statusMessage = (_lastProgressStatus == ProgressStatus.Paused) ? ServicesVSResources.Paused_0_tasks_in_queue : ServicesVSResources.Evaluating_0_tasks_in_queue;
-                return string.Format(statusMessage, _lastPendingItemCount);
-            }
-
-            void EnqueueRefresh()
-            {
-                if (_resettableDelay != null)
-                {
-                    _resettableDelay.Reset();
-                    return;
-                }
-
-                _resettableDelay = new ResettableDelay((int)s_minimumInterval.TotalMilliseconds, AsynchronousOperationListenerProvider.NullListener);
-                _resettableDelay.Task.SafeContinueWith(_ =>
-                {
-                    _resettableDelay = null;
-                    ProgressUpdated();
-                }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
-            }
+            var statusMessage = (progressData.Status == ProgressStatus.Paused) ? ServicesVSResources.Paused_0_tasks_in_queue : ServicesVSResources.Evaluating_0_tasks_in_queue;
```

Comment:
nit1: parens not needed.
nit2: wrap long line like so:

```c#
var statusMessage = condition
    ? Paused
    : Evaluating
```

Reply:
nit3: this helper doesn't seem... helpful :) just inline this at the places you are calling the helper. you already know the status there, so htere's no need to check it here.